main.tf

```diff
@@ -191,6 +191,18 @@ resource "azurerm_application_gateway" "ag" {
     }
   }
 
+  dynamic "ssl_policy" {
+    for_each = var.ssl_policy != null ? [var.ssl_policy] : []
+    content {
+      disabled_protocols   = var.ssl_policy.policy_type == null && var.ssl_policy.policy_name == null ? var.ssl_policy.disabled_protocols : null
```

Comment:
are all nulls needed if the top-level dynamic is null?

Reply:
Yes - if we do not have default null for objects, we will have to pass something from the calling repository. This way, we can only pass value of the `policy_name` and `min_protocol_version`  from calling repo.